receipt.go

```diff
@@ -68,11 +122,206 @@ type ReceiptFullInfo struct {
+		return 0, fmt.Errorf("error creating request: %s", err.Error())
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		errorResp := errorResponse{}
+		if err = json.Unmarshal(resp.Body, &errorResp); err != nil {
+			return 0, fmt.Errorf("error unmarshalling error response: %s responseBody: %s",
+				err.Error(),
+				string(resp.Body),
+			)
+		}
+		return 0, fmt.Errorf("failed to get txID: %s", errorResp.Reason)
+	}
+
+	// Parse the response
+	var txID int64
+	err = json.Unmarshal(resp.Body, &txID)
+	if err != nil {
+		return 0, fmt.Errorf("error unmarshalling response: %s", err.Error())
+	}
+
+	return txID, nil
+}
+
+func (r *receipt) RegisterTXID(ctx context.Context, txID int64) (ReceiptInfo, error) {
+
+	txIDInfo := txIDReq{TXID: txID}
+	reqBody, err := json.Marshal(txIDInfo)
+	if err != nil {
+		return ReceiptInfo{}, fmt.Errorf("error marshalling request body: %s", err.Error())
+	}
+	endpoint := r.gateway.FactoryEndpoint(receiptRegisterTXIDEndpoint)
+	resp, err := r.gateway.HTTPRequest(
+		ctx,
+		endpoint,
+		http.MethodPost,
+		constants.ContentTypeUrlEncoded,
+		reqBody,
+		nil,
+	)
+	if err != nil {
+		return ReceiptInfo{}, fmt.Errorf("error creating request: %s", err.Error())
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		errorResp := errorResponse{}
+		if err = json.Unmarshal(resp.Body, &errorResp); err != nil {
+			return ReceiptInfo{}, fmt.Errorf("error unmarshalling error response: %s responseBody: %s",
+				err.Error(),
+				string(resp.Body),
+			)
+		}
+		return ReceiptInfo{}, fmt.Errorf("failed to register txID: %s", errorResp.Reason)
+	}
+
+	receiptInfo := ReceiptInfo{}
+	err = json.Unmarshal(resp.Body, &receiptInfo)
+	if err != nil {
+		return ReceiptInfo{}, fmt.Errorf("error unmarshalling response: %s", err.Error())
+	}
+
+	return receiptInfo, nil
+}
+
+func (r *receipt) GetReceiptInfo(ctx context.Context, index uint32) (ReceiptFullInfo, error) {
+	bodyBytes, err := json.Marshal(indexInfo{Index: index})
+	if err != nil {
+		return ReceiptFullInfo{}, fmt.Errorf("error marshalling body: %s", err.Error())
+	}
+	endpoint := r.gateway.FactoryEndpoint(receiptInfoEndpoint)
+	resp, err := r.gateway.HTTPRequest(
+		ctx,
+		endpoint,
+		http.MethodPost,
+		constants.ContentTypeUrlEncoded,
+		bodyBytes,
+		nil,
+	)
+	if err != nil {
+		return ReceiptFullInfo{}, fmt.Errorf("error creating request: %s", err.Error())
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		errorResp := errorResponse{}
+		if err := json.Unmarshal(resp.Body, &errorResp); err != nil {
+			return ReceiptFullInfo{}, fmt.Errorf("error unmarshalling error response: %s responseBody: %s",
+				err.Error(),
+				string(resp.Body),
+			)
+		}
+
+		return ReceiptFullInfo{}, fmt.Errorf("failed to get Z report info: %s", errorResp.Reason)
+	}
+
+	receiptFullInfo := ReceiptFullInfo{}
+	if err := json.Unmarshal(resp.Body, &receiptFullInfo); err != nil {
+		return ReceiptFullInfo{}, fmt.Errorf("error unmarshalling response: %s", err.Error())
+	}
+	return receiptFullInfo, nil
```

Comment:
we should beconsistent in our code style, if we are making break lines before `return` we are making them every method. if's is also a matter of line break at least after logic block ends. ofc this could be fixed easily by gofmt formatter

Reply:
fixed